google/generativeai/generative_models.py

```diff
@@ -207,31 +205,33 @@ def from_cached_content(
     def from_cached_content(
         cls,
         cached_content: str | caching.CachedContent,
+        *,
         generation_config: generation_types.GenerationConfigType | None = None,
         safety_settings: safety_types.SafetySettingOptions | None = None,
     ) -> GenerativeModel:
         """Creates a model with `cached_content` as model's context.
 
         Args:
             cached_content: context for the model.
+            generation_config: Overrides for the model's generation config.
+            safety_settings: Overrides for the model's safety settings.
 
         Returns:
             `GenerativeModel` object with `cached_content` as its context.
         """
         if isinstance(cached_content, str):
             cached_content = caching.CachedContent.get(name=cached_content)
 
-        # call __new__ with the cached_content to set the model's context. This is done to avoid
-        # the exposing `cached_content` as a public attribute.
-        self = cls.__new__(cls, cached_content=cached_content)
-
         # call __init__ to set the model's `generation_config`, `safety_settings`.
         # `model_name` will be the name of the model for which the `cached_content` was created.
-        self.__init__(
+        self = cls(
             model_name=cached_content.model,
             generation_config=generation_config,
             safety_settings=safety_settings,
         )
+
+        # set the model's context.
+        setattr(self, "_cached_content", cached_content.name)
```

Comment:
Why is this done with `setattr`? Dynamic properties can mess with type checking, so preferably this should just be a normal instance field.

If there's a good reason for it, we should document it somewhere around here too.

Reply:
Should I change it? Reason was:

`GenerativeModel` instances created using the factory method `from_cached_content` will have `_cached_content` property, instances created using normal `__init__` would not have this attr, this is handled by the public property which will return None in latter case.

Reply:
Seems like it could still have `None` for regular instances? Not a biggy though, it's internal only so let's get this in first.